ncm-cron/src/main/pan/components/cron/schema.pan

```diff
@@ -38,15 +38,103 @@ type structure_cron_log = {
     'mode' ? string
 } with structure_cron_log_valid(SELF);
 
+@documentation{
+    Validate contents of cron timing fields (see CRONTAB(5) for details)
+
+    Cron timing fields can contain complex expressions (e.g. "1,5,13-23/2"). Rather than validate these in
+    depth the aim here is to catch things that are obviously wrong, such as:
+        * characters which are not valid in cron fields
+        * out of range numbers (e.g. "35" in the hour field)
+        * names in the wrong field (e.g. "tue" in the day of month field)
+}
+function valid_cron_timing = {
+    if (ARGC != 4) {
+        error(format('%s: expected 4 parameters, received %d', FUNCTION, ARGC));
+    };
+
+    timing_value = to_lowercase(ARGV[0]);
+    lower_bound = ARGV[1];
+    upper_bound = ARGV[2];
+    text_regex = ARGV[3];
+
+    # Check that the field contains only valid characters
+    if (!match(timing_value, '^(?:[a-z0-9/*-,]+)$')) error(format('"%s" contains invalid characters', timing_value));
+
+    # Find runs of digits and validate them against provided bounds
+    foreach(k; v; matches(timing_value, '([0-9]+)')) {
+        i = to_long(v);
+        if (i < lower_bound) error(format('Value %d is below lower bound of %d', i, lower_bound));
+        if (i > upper_bound) error(format('Value %d is above upper bound of %d', i, upper_bound));
+    };
+
+    # Find runs of letters and validate them against provided regex
+    foreach(k; v; matches(timing_value, '([a-z]+)')) {
+        if (!match(v, text_regex)) error(format('"%s" is not a valid value for this field item', v));
+    };
+
+    # Ignore all other characters
+    true;
+};
+
+@documentation{ Convenience wrapper for validating cron minute field }
+function valid_cron_minute = valid_cron_timing(ARGV[0], 0, 59, '^(?![a-z]).+$');
```

Comment:
ah, more test_is_type crap, we do need https://github.com/quattor/pan/pull/123 .....

Reply:
Indeed, these would be types if they weren't re-used to check the `frequency` field.

Reply:
or if we had a panc release with the validate function.... grrrr